### DISCUSSION TITLE
ESP32 - S3 board for your lesson

### DISCUSSION THREAD

**Discussion by mightylastingcode** [Mair/esp32-course#65] [Course Qeustions] [answered]
Hi Mair,

Thank you for adding new lessons.

This time I want to try the lessons with the new ESP32 S3 DevKitC 1-N-8R8 board with 8M Flash and 8M RAM.    It is good for building a bigger project.

But, I have trouble with flashing the device.    I am using IDF 4.4 version.   I have even tried the master version.     Have you used this new device yet?   

Thank you if you can help.
Michael

PS : Running IDF tool on Ubuntu Linux 20.04 LT

=========================  Log ==============================================

esptool.py esp32s3 -p /dev/ttyUSB0 -b 460800 --before=default_reset --after=hard_reset write_flash --flash_mode dio --flash_freq 80m --flash_size 2MB 0x0 bootloader/bootloader.bin 0x10000 hello-world.bin 0x8000 partition_table/partition-table.bin
esptool.py v3.2-dev
Chip is ESP32-S3
Configuring flash size...
Flash will be erased from 0x00000000 to 0x00004fff...
Flash will be erased from 0x00010000 to 0x00039fff...
Flash will be erased f
rom 0x00008000 to 0x00008fff...
Compressed 18896 bytes to 11765...
Writing at 0x00000000... (100 %)

A fatal error occurred: Invalid head of packet (0x1B)
CMake Error at run_serial_tool.cmake:50 (message):
/home/michael/.espressif/python_env/idf4.4_py3.8_env/bin/python
/home/michael/esp_v44b/esp-idf/components/esptool_py/esptool/esptool.py
--chip esp32s3 failed

FAILED: CMakeFiles/flash
cd /home/michael/esp_v44b/esp-idf/components/esptool_py && /usr/bin/cmake -D IDF_PATH="/home/michael/esp_v44b/esp-idf" -D

**Comment by Mair** [comment]
@mightylastingcode 

 Not sure but I was wandering if you set your target to the S3.
try 
```bash
idf.py set-target esp32s3
```
then flash your chip.

Just a note of caution. I have found setting a target find of resets your config so be sure to make a mental note of what config you need to change back if any

**Reply by mightylastingcode** [reply]
Thank you for your reply.   I did.   Sadly, it did not work.    Are you saying that if I set the target to esp32 with another project, I need to do this target device config  back to S3 again with my S3 project?

I am still waiting for Espressif to respond to my new issue post.  Hope they can help me.   The wait is pretty long.

Thank you. for trying to help me!    
Michael

**Reply by Mair** [reply]
> Thank you for your reply. I did. Sadly, it did not work. Are you saying that if I set the target to esp32 with another project, I need to do this target device config back to S3 again with my S3 project?

I could be wrong but I believe its project specific and targets the munuconfig 

> Sadly, it did not work

That's too bad. It seems like it is able to compile but unable to find the chip to flash. maybe a driver issue for the USB to UART bridge on your dev board?
what does your device manager say?

**Reply by mightylastingcode** [reply]
Hi Mair,

Thank you for sharing your idea.   I am not sure.

If it were the Linux uart driver issue, I think that I would not be able to use the serial monitor.  The serial monitor seems to work well with my S3 device.    

As you can see, I can get the logging message from the chip.    I think that the UART driver is ok.   The info from the chip is a little suspicious.  It said that I only have 2MB.  I have 8MB on board.   The date is Mar 27 2021.   I heard that there are some beta chips out there that may have some flashing issues.  

Up to now, my post on Espressif's forum has been viewed by many people.  But, no one has replied yet.  I may take the issue to Adafruit since I purchased the device from them.  They have successfully developed their CircuitPython to run on S3 and demostrated it in their Youtube channel.   I hope that they may know the underlying issue and may be willing to share it with me.

Adafruit staff were using Windows 10.   I may try it with Windows 10 and see if it makes a difference.  Maybe the Window UART driver is better.

Here is what I get from my S3 chip's USB to UART port.    There are a lot of info.  I just show you a portion here.

michael@michael-Inspiron-7472:~/Documents/ESP32_IDF44b_projects/examples/esp32_s3_devkitc1/get-started/hello_world$ idf.py -p /dev/ttyUSB0 monitor

....


I (478) TinyUSB: TinyUSB Driver installed
I (488) example: USB initialization DONE
SBuild:Mar 27 2021
device reports readiness to read but returned no data (device disconnected or multiple access on port?)
rst:01 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOTc8
load:0x403b6000,len:0x8c8
load:0x403ba000,len:0x2c04
entWaiting for the device to reconnect
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x1 (POWERON),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
mode:DIO, clock div:1
load:0x3fcd0108,len:0x15c8
load:0x403b6000,len:0x8c8
load:0x403ba000,len:0x2c04
entry 0x403b61c4
I (24) boot: ESP-IDF v4.4-dev-2594-ga20df743f1-dirty 2nd stage bootloader
I (25) boot: compile time 16:57:21
I (25) boot: chip revision: 0
I (28) boot.esp32s3: SPI Speed : 80MHz
I (33) boot.esp32s3: SPI Mode : DIO
I (38) boot.esp32s3: SPI Flash Size : 2MB                 <<< Not True!!!  It should be 8MB!>>
I (43) boot: Enabling RNG early entropy source...
W (48) bootloader_random: RNG for ESP32-S3 not currently supported
I (55) boot: Partition Table:
I (59) boot: ## Label Usage Type ST Offset Length
I (66) boot: 0 nvs WiFi data 01 02 00009000 00006000
I (73) boot: 1 phy_init RF data 01 01 0000f000 00001000
I (81) boot: 2 factory factory app 00 00 00010000 00100000
I (88) boot: End of partition table
I (92) esp_image: segment 0: paddr=00010020 vaddr=3c020020 size=090e0h ( 37088) map
I (108) esp_image: segment 1: paddr=00019108 vaddr=3fc90dd0 size=0253ch ( 9532) load
I (111) esp_image: segment 2: paddr=0001b64c vaddr=40374000 size=049cch ( 18892) load
I (122) esp_image: segment 3: paddr=00020020 vaddr=42000020 size=1b660h (112224) map
I (147) esp_image: segment 4: paddr=0003b688 vaddr=403789cc size=08404h ( 33796) load
I (154) esp_image: segment 5: paddr=00043a94 vaddr=50000000 size=00010h ( 16) load

**Reply by Mair** [reply]
@mightylastingcode you are correct it does not seem to be a driver issue. I have an s3 I can flash with no issue so it is strange.

> It said that I only have 2MB

You need to set the flash size in the menuconfig under serial flasher config

![image](https://user-images.githubusercontent.com/2477839/161883727-ebd04ac2-6bde-4188-8b38-2d738f2e9078.png)

**Reply by mightylastingcode** [reply]
Awesome.  So good to hear that you can flash your S3 device.   Yes, you are right that you can change the flash size.  Yes, I can do that for mine once I can flash my device.   

I would like to know your setup.  I want to try it the way you did it.

1. OS : Window 10 or Linux
2. IDF version
3. idf.py or VSC plug in

It may be possible that I just have a bad part.  I remember that a few years ago some ESP32 parts required to hold down the boot button to flash the devices.    I will try to contact Adafruit support line for help too.

Thank you for helping me.    
Michael

**Comment by Mair** [comment, marked as answer]
> Awesome. So good to hear that you can flash your S3 device. Yes, you are right that you can change the flash size. Yes, I can do that for mine once I can flash my device.
